### PR TITLE
⬆️ Upgrade package to work with ESLint 8

### PR DESCRIPTION
**Pull request by wcandillon**
## Summary

This are the two package upgrade required for this package to run with ESLint 8

## Changelog

[JavaScript] [Changed] `@react-native-community/eslint-config` to work with ESLInt 8

## Test Plan

Try the package with ESLint 8

### CODE REVIEW

File: packages/eslint-config-react-native-community/package.json [hunk 1]
```diff
@@ -12,12 +12,12 @@
   "homepage": "https://github.com/facebook/react-native/tree/HEAD/packages/eslint-config-react-native-community#readme",
   "dependencies": {
     "@react-native-community/eslint-plugin": "^1.1.0",
-    "@typescript-eslint/eslint-plugin": "^4.22.1",
+    "@typescript-eslint/eslint-plugin": "^5.15.0",
     "@typescript-eslint/parser": "^4.22.1",
```
Comment (Kerumen) [inline packages/eslint-config-react-native-community/package.json:16]: `@typescript-eslint/eslint-plugin` and `@typescript-eslint/parser` should be on the same version

Comment (wcandillon) [reply packages/eslint-config-react-native-community/package.json:16]: thks for catching this

**Comment by robrechtme** [comment]
Maybe also change the `peerDependencies` to ` "eslint": "^7 || ^8"`?

**Comment by wcandillon** [comment]
@robrechtme done 👍🏻

File: packages/eslint-config-react-native-community/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@react-native-community/eslint-config",
-  "version": "3.0.1",
+  "version": "3.0.2",
```
Comment (kelset) [inline packages/eslint-config-react-native-community/package.json:3]: shouldn't this be a more "important" bump? Like, adding support for ESLint 8 seems something that should be at least a minor bump if not even a major 🤔

Comment (wcandillon) [reply packages/eslint-config-react-native-community/package.json:3]: This is not a breaking change but maybe 3.1.0?

**Comment by wcandillon** [comment]
Would it be possible to publish a new version of this package?

**Comment by cortinico** [comment]
> Would it be possible to publish a new version of this package?

@matt-oakes should have push rights for this package. Could you help here?

**Pull request closed without merging**